addons/account_payment/controllers/portal.py

```diff
@@ -52,8 +52,12 @@ def _invoice_get_page_view_values(self, invoice, access_token, **kwargs):
                 providers_sudo
             ),
         }
+        amount = invoice.amount_residual
+        if invoice._is_eligible_for_early_payment_discount(invoice.currency_id, fields.Date.context_today(invoice)):
```

Comment:
why don't you do an helper function to compute the total residual invoice amount to pay for a given payment date? You are computing multiple times the same thing using different ways :face_with_spiral_eyes: